src/lib/components/FornaContainer.react.js

```diff
@@ -0,0 +1,206 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import {FornaContainer as PreFornaContainer} from 'fornac';
+
+/**
+ * This is a FornaContainer component.
+ */
+export default class FornaContainer extends Component {
+    constructor(props) {
+        super(props);
+        this.renderNewSequences = this.renderNewSequences.bind(this);
+    }
+
+    componentDidMount() {
+        const {id, height, width} = this.props;
+
+        this._fornaContainer = new PreFornaContainer('#' + id, {
+            initialSize: [height, width],
+        });
+
+        this.renderNewSequences();
+    }
+
+    renderNewSequences() {
+        const {sequences} = this.props;
+
+        if (this._fornaContainer) {
+            this._fornaContainer.clearNodes();
+
+            sequences.forEach(seq => {
+                const unpackedOptions = Object.assign(
+                    seq.options ? seq.options : {},
+                    {sequence: seq.sequence, structure: seq.structure}
+                );
+                console.warn(unpackedOptions);
+                this._fornaContainer.addRNA(seq.structure, unpackedOptions);
+            });
+        }
+    }
+
+    shouldComponentUpdate(nextProps) {
+        const {sequences, nodeFillColor, colorScheme} = this.props;
+
+        // update the component if the actual sequences have changed
+        if (
+            (sequences && !nextProps.sequences) ||
+            (!sequences && nextProps.sequences) ||
+            sequences.length !== nextProps.sequences.length ||
+            sequences.some(
+                (seq, i) =>
+                    sequences[i].sequence !== nextProps.sequences[i].sequence ||
+                    sequences[i].structure !==
+                        nextProps.sequences[i].structure ||
+                    (sequences[i].options && !nextProps.sequences[i].options) ||
+                    (!sequences[i].options && nextProps.sequences[i].options) ||
```

Comment:
A pattern I like for `(a && !b) || (!a && b)` is `(Boolean(a) !== Boolean(b))` (or `(!!a !== !!b)` but I some of our linters disapprove of `!!`)

Also https://ramdajs.com/docs/#zip would likely clean this section up a lot